### ISSUE TITLE
E0603 (function import is private) does not detected 

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#8500]
## Environment

* **IntelliJ Rust plugin version:** 0.4.165.4410-213-nightly
* **Rust toolchain version:** 1.60.0-nightly (5e57faa78 2022-01-19) x86_64-apple-darwin
* **IDE name and version:** CLion 2022.1 EAP (CL-221.3427.90)
* **Operating system:** macOS 11.4
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description
<img width="601" alt="image" src="https://user-images.githubusercontent.com/49211026/152155208-ae66d70e-a766-418e-acbd-82a0b4332308.png">

## Steps to reproduce
```rust
fn main() {
      a::foo(); // error 
}

mod a {
    use b::foo;

    pub(crate) mod b {
        pub fn foo() {}
    }
}
```
